pagopa/io-app
# feat: [IOAPPCOM-44, IOAPPCOM-47, IOAPPCOM-49, IOAPPCOM-52] Support for attachments on standard messages

## Short description
This PR adds support for PDF attachments on standard messages (which are not medical prescriptions).

## List of changes proposed in this pull request
In order not to break the existing PN flow (since it slightly differs from generic attachments one), the following changes have been made:
- In-app model of the attachment has a category, to distinguish it between generic, PN and MVL attachment (even though the last should be removed in the future)
- Standard message detail component retrieves the third-party message list of attachments (if there is one) upon first loading, while rendering the markdown. It then makes sure to render the list only after the markdown has been loaded.
- Reusable PN hook for showing the bottom sheet has been edited in order not to start the attachment download automatically, but just notifying the calling component of a single attachment request from the user (after having shown the alert on the bottom sheet - if the preference allows it)
- The Attachment preview component downloads the attachment if it is a generic one (and has to be downloaded) and handles the proper loading or errors

This PR does not include mixpanel events, which will be added in a later one

## How to test
You can either use the dev-server (`withRemoteAttachments` setting in the `config`) or use the test credentials to try it out with the `io-backend` (contact me or the red unit to retrieve them).
While medical prescription cannot be tested, you can check that standard (non third-party messages) still work (they do not have attachments), that PN messages with and without attachments still work and that third-party messages with attachment are now correctly supported.
Make also sure to test the three main download errors (a failing download, an invalid PDF and a corrupted one) as well as cancelling a (generic third-party attachment) download ([reference FIGMA here](https://www.figma.com/file/cXzK7qaiagRil5vZy6EqQS/Messaggi-con-allegati_Errori?node-id=0%3A1&t=RB2fB8pcBZAdxofm-0)).
